src/xrpld/app/tx/detail/Transactor.cpp

```diff
@@ -133,6 +133,33 @@ preflight1(PreflightContext const& ctx)
 NotTEC
 preflight2(PreflightContext const& ctx)
 {
+    if (ctx.flags & tapDRY_RUN)  // simulation
+    {
+        if (ctx.tx.getSignature().empty())
+        {
+            if (ctx.tx.isFieldPresent(sfSigners))
+            {
+                for (auto const& signer : ctx.tx.getFieldArray(sfSigners))
+                {
+                    if (signer.isFieldPresent(sfTxnSignature) &&
+                        !ctx.tx.getSignature().empty())
+                    {
+                        // NOTE: This code should never be hit because it's
+                        // checked in the `simulate` RPC
+                        return temINVALID;  // LCOV_EXCL_LINE
+                    }
+                }
+                return tesSUCCESS;
+            }
+            else
```

Comment:
`else` block is excessive here, just `return tesSUCCESS;` will be ok

Reply:
fixed in 316639642e6a4d8111a5da29af35c965dbc11d01